app/views/shared/means/_cash_income.html.erb

```diff
@@ -26,7 +26,7 @@
       </div>
 
       <%= form.govuk_radio_divider %>
-      <%= form.govuk_check_box :none_selected, true, "", multiple: false, label: { text: none_of_the_above }, checked: @legal_aid_application.no_cash_income? %>
+      <%= form.govuk_check_box :none_selected, true, "", multiple: false, label: { text: none_of_the_above }, checked: type.present? ? type.no_cash_income? : @legal_aid_application.no_cash_income? %>
```

Comment:
As you have updated the means versions to set type as @legal_aid_application you can remove the ternary check and just use 
```suggestion
      <%= form.govuk_check_box :none_selected, true, "", multiple: false, label: { text: none_of_the_above }, checked: type.no_cash_income? %>
```